mancala/server/src/lib.rs

```diff
@@ -1,24 +1,36 @@
 use std::io::{Read, Write};
 use std::net::{Shutdown, TcpListener, TcpStream};
+use std::str;
 use std::thread;
 
 fn handle_client(mut stream: TcpStream) {
-    let mut data = [0 as u8; 50]; // using 50 byte buffer
-    while match stream.read(&mut data) {
-        Ok(size) => {
-            // echo everything!
-            stream.write_all(&data[0..size]).unwrap();
-            true
-        }
-        Err(_) => {
-            println!(
-                "An error occurred, terminating connection with {}",
-                stream.peer_addr().unwrap()
-            );
-            stream.shutdown(Shutdown::Both).unwrap();
-            false
+    let mut buffer = [0; 512]; // use a 512 byte buffer
+    loop {
+        match stream.read(&mut buffer) {
+            Ok(size) => {
+                let input = str::from_utf8(&buffer[0..size]).unwrap().trim_end();
+                println!("Data received: {}", input);
+                // Exit loop & terminate connection if user enters "quit"
+                if input.eq_ignore_ascii_case("quit") {
+                    println!("Client terminated connection");
+                    stream.shutdown(Shutdown::Both).unwrap();
+                    break;
+                } else {
+                    // TODO: update & send game state instead of echoing
+                    stream.write_all(&buffer[0..size]).unwrap();
+                    stream.flush().unwrap();
+                }
+            }
+            Err(_) => {
+                println!(
+                    "An error occurred, terminating connection with {}",
+                    stream.peer_addr().unwrap()
+                );
+                // Close read & write portions of connection
+                stream.shutdown(Shutdown::Both).unwrap();
+            }
```

Comment:
easy fix to above problem is just adding a return statement here

```
// Close read & write portions of connection
stream.shutdown(Shutdown::Both).unwrap();
return
```

Reply:
actually, just tried this and it doesn't work, the server continues to loop "data received". I'll look at alternatives

Reply:
hmmm, works for me...? lets check it out later maybe?

![client_server](https://user-images.githubusercontent.com/33067676/75074352-54171e80-54b0-11ea-9e9c-ed4b64e63764.gif)

Reply:
strange! either way, I figured an alternative that works. Check it out in the commit

Reply:
![Screen-Recording-2020-02-21-at-2](https://user-images.githubusercontent.com/38337779/75075808-7bbbb600-54b3-11ea-8310-f3d94868d352.gif)

Reply:
nope. the return alone isn't doing it.